salt/modules/junos.py

```diff
@@ -1310,3 +1320,117 @@ def commit_check():
+    if key_items is not None:
+        get_kvargs['key_items'] = key_items
+    if filters is not None:
+        get_kvargs['filters'] = filters
+    if args is not None and isinstance(args, dict):
+        get_kvargs['args'] = args
+    pyez_tables_path = os.path.dirname(os.path.abspath(tables_dir.__file__))
+    try:
+        file_loc = glob.glob(os.path.join(path, '{}'.format(file))) or \
+                   glob.glob(os.path.join(pyez_tables_path, '{}'.format(file)))
+        if len(file_loc) == 1:
+            file_name = file_loc[0]
+        elif len(file_loc) > 1:
+            ret['message'] = 'Given table file %s is located at multiple location'\
+                % file
+            ret['out'] = False
+            return ret
+        elif len(file_loc) == 0:
+            ret['message'] = 'Given table file %s cannot be located' % file
+            ret['out'] = False
+            return ret
+        try:
+            with salt.utils.fopen(file_name) as fp:
+                ret['table'] = yaml.load(fp.read(),
+                                         Loader=yamlordereddictloader.Loader)
+                globals().update(FactoryLoader().load(ret['table']))
+        except IOError as err:
+            ret['message'] = 'Uncaught exception during YAML Load - please ' \
+                             'report: {0}'.format(str(err))
+            ret['out'] = False
+            return ret
+        try:
+            data = globals()[table](conn)
+            data.get(**get_kvargs)
+        except KeyError as err:
+            ret['message'] = 'Uncaught exception during get API call - please ' \
+                             'report: {0}'.format(str(err))
+            ret['out'] = False
+            return ret
+        ret['reply'] = json.loads(data.to_json())
+        if data.__class__.__bases__[0] == OpTable:
+            # Sets key value if not present in YAML. To be used by returner
+            if ret['table'][table].get('key') is None:
+                ret['table'][table]['key'] = data.ITEM_NAME_XPATH
+            # If key is provided from salt state file.
+            if key is not None:
+                ret['table'][table]['key'] = data.KEY
+        else:
+            if target is not None:
+                ret['table'][table]['target'] = data.TARGET
+            if key is not None:
+                ret['table'][table]['key'] = data.KEY
+            if key_items is not None:
+                ret['table'][table]['key_items'] = data.KEY_ITEMS
+            if args is not None:
+                ret['table'][table]['args'] = data.CMD_ARGS
+                ret['table'][table]['command'] = data.GET_CMD
+    except ConnectClosedError:
+        ret['message'] = 'Got ConnectClosedError exception. Connection lost ' \
+                         'with %s' % conn
+        ret['out'] = False
+        return ret
+    except Exception as err:
+        ret['message'] = 'Uncaught exception - please report: {0}'.format(
+            str(err))
```

Comment:
Does this `str()` type work with Python 3? I think that should be `six.text_type`.